src/components/IconCard/icon-card.scss

```diff
@@ -1,27 +1,119 @@
-.icon-container {
+.icons-list {
   display: flex;
-  flex-flow: row wrap;
+  flex-flow: wrap;
   width: 100%;
   padding-bottom: $spacing-06;
 }
 
-.icon-search {
-  margin-bottom: rem(32px);
-}
-
-.icon {
-  flex: 1 1 10rem;
-  max-width: 10rem;
+.icon__container {
+  flex: 0 1 50%;
   position: relative;
 
   &:nth-child(n) {
+    border-top: 1px solid $carbon--gray-20;
     border-left: 1px solid $carbon--gray-20;
-    border-bottom: 1px solid $carbon--gray-20;
+  }
+  &:nth-child(n):nth-child(-n + 2) {
+    border-top: none;
+  }
+  &:nth-child(2n - 1) {
+    border-left: none;
+  }
+
+  & .bx--aspect-ratio > * {
+    z-index: 100; // all siblings of bx--aspect-ratio--object will have same z-index
+  }
+}
+
+@media all and (min-width: 515px) {
+  .icon__container {
+    flex: 0 1 calc(100% / 3);
+
+    &:nth-child(n) {
+      border-top: 1px solid $carbon--gray-20;
+      border-left: 1px solid $carbon--gray-20;
+    }
+    &:nth-child(n):nth-child(-n + 3) {
+      border-top: none;
+    }
+    &:nth-child(3n - 2) {
+      border-left: none;
+    }
+  }
+}
+
+@media all and (min-width: 730px) {
```

Comment:
The styles and media queries here seem very specific and potentially fragile with respect to minor tweaks. Would we want to explore another way of achieving this layout? Not quite sure what the rules are but would be worried about including one-off breakpoints for this. 

Is the sketch kit itself aligned to the grid? Maybe we could use the column classes for this? If not, I wonder if we could use something like CSS Grid to help us out here 🤔 cc @alisonjoseph is that something we can use on the site?

Reply:
We can't use the column classes easily here until we switch this site over to the Gatsby theme / get rid of the offset for each row. (which I already did over on the IDL site https://github.com/carbon-design-system/design-language-website/blob/master/src/components/IconLibrary/icon-library.scss)

I think we can leave it for now, and hopefully switch it when we do the birthday cake version of this site.

Reply:
this is a refactoring of previously-existing CSS, so it isn't a departure from what's live on the current site. the only change is how sizing & flex works for each displayed icon, so that they visually align to edges of their container (instead of being constrained to a maximum width). 

and yeah as alison said we can totally refactor this when merged with the new gatsby theme  :)

Reply:
Sounds good!